rolesPath not being captured correctly in Auth Config

Here is my server config 

```ts
const neoSchema = new Neo4jGraphQL({
  typeDefs,
  resolvers,
  driver,
  config: {
    jwt: {
      secret: process.env.JWT_SECRET,
      rolesPath: process.env.AUTH_DIRECTIVES_ROLE_KEY,
    },
  },
})
```

I have an env variable with my auth directive role key `AUTH_DIRECTIVES_ROLE_KEY=https://flcadmin.netlify.app/roles`


This is what my params object looks like 

```
{
   "auth": {
     "isAuthenticated": true,
     "jwt": {
       "https://flcadmin.netlify.app/roles": [
         "federalAdmin"
       ],
       "iss": "https://flcadmin.us.auth0.com/",
       "sub": "auth0|6089feb8e1e3e700697f7eff",
       "aud": [
         "https://flcadmin.netlify.app/graphql",
         "https://flcadmin.us.auth0.com/userinfo"
       ],
       "iat": 1623180461,
       "exp": 1623266861,
       "azp": "dD6nn8AsoPL4PqKTqwG9vTHjDnSlRIqW",
       "scope": "openid profile email",
       "permissions": []
     }
   }
 }
```

And here is my auth directive in my schema.graphql 

```gql
extend type Member
  @auth(rules: [{ operations: [READ],  roles: ["federalAdmin"]  }])
```


when I run the query 

```gql
{
  members {
    id
    firstName
    lastName
  }
}
```

I get this error. Any help as to what I'm doing wrongly?

``` "Failed to invoke procedure `apoc.util.validate`: Caused by: java.lang.ClassCastException: class org.neo4j.values.storable.NoValue cannot be cast to class org.neo4j.values.storable.BooleanValue (org.neo4j.values.storable.NoValue and org.neo4j.values.storable.BooleanValue are in unnamed module of loader 'app')"```

I noticed that if I set other keys like "permissions", or "scope" it is captured correctly, so I'm using that as a temporary workaround for now. I think it has to do with the fact that the key is a url.

**System**

-   OS: macOS
-   Version: @neo4j/graphql@1.0.2
-   Node.js version: 14.16.0

@danstarns This is what was logged when I ran the query. Does it help?

```
2021-06-23T13:36:48.531Z @neo4j/graphql:graphql Incoming GraphQL:
 Query:
 {
   members {
     id
     firstName
     lastName
   }
 }
 Variables:
 {}
 2021-06-23T13:36:48.534Z @neo4j/graphql:auth Verifying JWT
 2021-06-23T13:36:48.538Z @neo4j/graphql:execute About to execute Cypher:
 Cypher:
 MATCH (this:Member)
 CALL apoc.util.validate(NOT(ANY(r IN ["federalAdmin"] WHERE ANY(rr IN $auth.roles WHERE r = rr))), "@neo4j/graphql/FORBIDDEN", [0])
 RETURN this { .id, .firstName, .lastName } as this
 Params:{
    "auth": {
      "isAuthenticated": true,
      "jwt": {
        "https://flcadmin.netlify.app/roles": [
          "federalAdmin"
        ],}
      }

      021-06-23T13:36:48.550Z @neo4j/graphql:execute Neo4jError: Failed to invoke procedure `apoc.util.validate`: Caused by: java.lang.ClassCastException: class org.neo4j.values.storable.NoValue cannot be cast to class org.neo4j.values.storable.BooleanValue (org.neo4j.values.storable.NoValue and org.neo4j.values.storable.BooleanValue are in unnamed module of loader 'app')
```

Here is additional debug info since I'm dealing with the same issue:

System

OS: WSL2
Version: @neo4j/graphql@2.4.0
Node.js version:  17.1.0

<pre>
@neo4j/graphql:graphql Incoming GraphQL:
Query:
{
  databases {
    name
  }
}
Variables:
{} +0ms
  graphql-parse-resolve-info [1] Entering fieldTreeFromAST with parent type 'Query' +0ms
  graphql-parse-resolve-info   [1] Processing AST 1 of 1; kind = Field +0ms
  graphql-parse-resolve-info   [1] Field 'databases' (alias = 'databases') +0ms
  graphql-parse-resolve-info   [1] Recursing into subfields +0ms
  graphql-parse-resolve-info     [2] Entering fieldTreeFromAST with parent type 'Database' +0ms
  graphql-parse-resolve-info       [2] Processing AST 1 of 1; kind = Field +0ms
  graphql-parse-resolve-info       [2] Field 'name' (alias = 'name') +1ms
  graphql-parse-resolve-info       [2] Exiting (no fields to add) +0ms
  @neo4j/graphql:auth Verifying JWT using secret +0ms
  @neo4j/graphql:execute About to execute Cypher:
Cypher:
MATCH (this:Database)
CALL apoc.util.validate(NOT(ANY(r IN ["federalAdmin"] WHERE ANY(rr IN $auth.roles WHERE r = rr))), "@neo4j/graphql/FORBIDDEN", [0])
RETURN this { .name } as this
Params:
{
  "auth": {
    "isAuthenticated": true,
    "jwt": {
      "https://data.academic.osisoft.com/roles": [
        "hub:read"
      ],
      "given_name": "Christian",
      "family_name": "Foisy",
      "nickname": "christian.foisy",
      "name": "Christian Foisy",
      "picture": "https://lh3.googleusercontent.com/a/AATXAJxE3aDGODmgkT9WN4OCAH0kWft81h6A0a5jGSVi=s96-c",
      "locale": "fr",
      "updated_at": "2021-11-24T18:56:36.244Z",
      "email": "christian.foisy@gmail.com",
      "email_verified": true,
      "iss": "https://dev-f0ejox1i.auth0.com/",
      "sub": "google-oauth2|112905675939324412337",
      "aud": "B4OT7ZWWLc8xXIkcynjyEfaIjRe2K2oH",
      "iat": 1637780196,
      "exp": 1637816196,
      "nonce": "xVvT31NOIInHQpwKRbEE",
      "org_id": "org_SiyOl6bv4Bl5occZ"
    }
  }
} +0ms
  @neo4j/graphql:execute Neo4jError: Failed to invoke procedure `apoc.util.validate`: Caused by: java.lang.ClassCastException: class org.neo4j.values.storable.NoValue cannot be cast to class org.neo4j.values.storable.BooleanValue (org.neo4j.values.storable.NoValue and org.neo4j.values.storable.BooleanValue are in unnamed module of loader 'app')

    at captureStacktrace (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/result.js:239:17)
    at new Result (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/result.js:59:23)
    at newCompletedResult (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/transaction.js:375:12)
    at Object.run (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/transaction.js:229:20)
    at Transaction.run (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/transaction.js:101:34)
    at /mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/@neo4j/graphql/dist/utils/execute.js:75:108
    at TransactionExecutor._safeExecuteTransactionWork (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:92:26)
    at TransactionExecutor._executeTransactionInsidePromise (/mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:83:34)
    at /mnt/c/Users/cfoisy/OneDrive - OSIsoft, LLC/Documents/Hub/github/academic-hub/graphql2/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:40:19
    at new Promise (<anonymous>) {
  constructor: [Function: Neo4jError],
  code: 'Neo.ClientError.Procedure.ProcedureCallFailed'
} +969ms

</pre>

--- a/docs/modules/ROOT/pages/auth/setup.adoc
+++ b/docs/modules/ROOT/pages/auth/setup.adoc
@@ -66,11 +66,13 @@ const neoSchema = new Neo4jGraphQL({
     config: {
         jwt: {
             jwksEndpoint: "https://YOUR_DOMAIN/.well-known/jwks.json",
-            rolesPath: "https://YOUR_DOMAIN/claims\\.https://YOUR_DOMAIN/claims/roles"
+            rolesPath: "https://auth0\\.mysite\\.com/claims.https://auth0\\.mysite\\.com/claims/roles"
         }
     }
 });
 ----
+
+Note that `.` characters within a key of the JWT must be escaped with `\\`, whilst a `.` character indicating traversal into a value must not be escaped.
 
 [[auth-setup-passing-in]]
 == Passing in JWTs

--- a/packages/graphql/tests/integration/directives/auth/roles.int.test.ts
+++ b/packages/graphql/tests/integration/directives/auth/roles.int.test.ts
@@ -954,4 +954,90 @@ describe("auth/roles", () => {
             }
         });
     });
+
+    describe("rolesPath with dots", () => {
+        test("can read role from path containing dots", async () => {
+            const session = driver.session();
+
+            const type = generateUniqueType("User");
+
+            const typeDefs = `
+                type ${type.name} {
+                    id: ID
+                    name: String
+                    password: String
+                }
+
+                extend type ${type.name}
+                    @auth(
+                        rules: [
+                            {
+                                roles: ["admin"]
+                            }
+                        ]
+                    )
+            `;
+
+            const userId = generate({
+                charset: "alphabetic",
+            });
+
+            const query = `
+                query {
+                    ${type.plural} {
+                        id
+                        name
+                        password
+                    }
+                }
+            `;
+
+            const neoSchema = new Neo4jGraphQL({
+                typeDefs,
+                config: {
+                    jwt: {
+                        secret,
+                        rolesPath: "https://auth0\\.mysite\\.com/claims.https://auth0\\.mysite\\.com/claims/roles",
+                    },
+                },
+            });
+
+            try {
+                await session.run(`
+                    CREATE (:${type.name} {id: "${userId}", name: "User1", password: "password" })
+                `);
+                // request without role "admin" - should return all users
+                const nonAdminReq = createJwtRequest(secret, {
+                    "https://auth0.mysite.com/claims": { "https://auth0.mysite.com/claims/roles": ["user"] },
+                    id: userId,
+                });
+
+                const gqlResultUser = await graphql({
+                    schema: neoSchema.schema,
+                    source: query,
+                    contextValue: { driver, req: nonAdminReq, driverConfig: { bookmarks: session.lastBookmark() } },
+                });
+
+                expect((gqlResultUser.errors as any[])[0].message).toBe("Forbidden");
+
+                // request with role "admin" - should return all users
+                const adminReq = createJwtRequest(secret, {
+                    "https://auth0.mysite.com/claims": { "https://auth0.mysite.com/claims/roles": ["admin"] },
+                    id: userId,
+                });
+
+                const gqlResultAdmin = await graphql({
+                    schema: neoSchema.schema,
+                    source: query,
+                    contextValue: { driver, req: adminReq, driverConfig: { bookmarks: session.lastBookmark() } },
+                });
+
+                expect(gqlResultAdmin.data).toEqual({
+                    [type.plural]: [{ id: userId, name: "User1", password: "password" }],
+                });
+            } finally {
+                await session.close();
+            }
+        });
+    });
 });

